File: C#/797_AllPathsFromSourceToTarget.cs
public class Solution {
    public IList<IList<int>> AllPathsSourceTarget(int[][] graph) {
        return Find(graph, new List<IList<int>>(), new List<int> {0}, 0);
    }
    
    public IList<IList<int>> Find(int[][] graph, IList<IList<int>> rst, List<int> curr, int index ) {
        foreach(int e in graph[index]) {
            List<int> aux = new List<int>(curr);
            aux.Add(e);
            if (e == graph.Length - 1) rst.Add(aux);
            else rst = Find(graph, rst, aux, e);
        }
        return rst;
    }
}

File: Python/797_AllPathsFromSourceToTarget.py
class Solution:
    def DFS(self, graph: List[List[int]], path: List[int], node: int) -> List[List[int]]:
        path.append(node)
        if node == len(graph) -1: return [path]
        if node == None: return []
        
        rtn = []
        for n in graph[node]:
            p = self.DFS(graph, path[:], n)
            if len(p) > 0: 
                rtn = rtn + p            
        return rtn
            
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        return self.DFS(graph, [], 0)
        